src/desert/_make.py

```diff
@@ -275,22 +288,15 @@ def field_for_schema(
             field = marshmallow.fields.Dict(
                 keys=field_for_schema(arguments[0]),
                 values=field_for_schema(arguments[1]),
+                **field_args,
             )
         elif typing_inspect.is_optional_type(typ):
             [subtyp] = (t for t in arguments if t is not NoneType)
             # Treat optional types as types with a None default
-            metadata[_DESERT_SENTINEL]["dump_default"] = metadata.get(
-                "dump_default", None
-            )
-            metadata[_DESERT_SENTINEL]["load_default"] = metadata.get(
-                "load_default", None
-            )
-            metadata[_DESERT_SENTINEL]["required"] = False
-
-            field = field_for_schema(subtyp, metadata=metadata, default=None)
-            field.dump_default = None
-            field.load_default = None
-            field.allow_none = True
+            metadata[_DESERT_SENTINEL]["allow_none"] = True
+            if default is marshmallow.missing:
```

Comment:
I would argue that this behavior is quite counter intuitive, but if we want to keep backward compatibility this is needed.

Reply:
If you want an optional value, then do the same as all python code and do `x: Optional[int] = None`